### PR TITLE
More optimised way of checking for medic weapons?

### PR DESCRIPTION
**Pull request by SimZor**
Resolves no issue

#### Changes proposed in this pull request: 
I'm not sure, but wouldn't this be a more optimised way of checking for a medic weapon? Instead of doing it in a loop, just checking when it's fired?

### CODE REVIEW

**Comment by BoGuu** [comment]
Think the idea was to stop them picking it up all together, i.e selling

**Comment by SimZor** [comment]
What about we do a check in onTakeItem aswell then? @BoGuu

**Comment by SimZor** [comment]
Mean either 😢

File: Altis_Life.Altis/core/fn_initMedic.sqf [hunk 1]
```diff
@@ -14,13 +14,18 @@ if ((FETCH_CONST(life_medicLevel)) < 1 && (FETCH_CONST(life_adminlevel) isEqualT
...
-            waitUntil {sleep 3; !(currentWeapon player isEqualTo "")};
-            removeAllWeapons player;
+    private _firedEVH = player addEventHandler ["Fired, {
+        // -- Get Params
+        params ["_unit", "", "", "_weapon"];
+        
+        // -- Check if weapon is equipped
+        if (!(_weapon _unit isEqualTo "")) then {
+            // -- Remove Weapon from Player
+            removeWeapon _weapon player;
```
Comment (BoGuu) [inline Altis_Life.Altis/core/fn_initMedic.sqf]: This can go again

Comment (SimZor) [inline Altis_Life.Altis/core/fn_initMedic.sqf]: Remove it all?

Comment (Jason2605) [inline Altis_Life.Altis/core/fn_initMedic.sqf]: I would guess, if you are restricting them picking the gun up, they won't be able to fire.

Comment (BoGuu) [inline Altis_Life.Altis/core/fn_initMedic.sqf]: As Jason said, we switched this to takeItem instead :)

File: Altis_Life.Altis/core/functions/fn_onTakeItem.sqf [hunk 2]
```diff
@@ -41,5 +41,14 @@ switch (playerSide) do
         if (_item in ["U_Rangemaster"]) then {
             [] call life_fnc_playerSkins;
         };
+
```
Comment (Jawshy) [inline Altis_Life.Altis/core/functions/fn_onTakeItem.sqf]: Please remove these spaces.

**Comment by SimZor** [comment]
Good to merge?

**Comment by Jawshy** [comment]
> Please remove these spaces.

**Comment by BoGuu** [comment]
LGTM

**Comment by BoGuu** [comment]
@SimZor _item is a classname already, can just remove the typeOf

**Comment by SimZor** [comment]
Sorry guys been busy, I'll commit when I get to work.

**Comment by SimZor** [comment]
fuck lets close this and make a new oen

**Pull request closed without merging**